// @docusaurus/plugin-ideal-image/lib/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const path_1 = __importDefault(require("path"));
function default_1(_context, options) {
    const isProd = process.env.NODE_ENV === 'production';
    return {
        name: 'docusaurus-plugin-ideal-image',
        getThemePath() {
            return path_1.default.resolve(__dirname, './theme');
        },
        configureWebpack(_config, isServer) {
            return {
                module: {
                    rules: [
                        {
                            test: /\.(png|jpe?g|gif)$/i,
                            use: [
                                '@endiliey/lqip-loader',
                                {
                                    loader: '@endiliey/responsive-loader',
                                    options: Object.assign({ emitFile: !isServer, disable: !isProd, 
                                        // eslint-disable-next-line
                                        adapter: require('@endiliey/responsive-loader/sharp'), name: isProd
                                            ? 'ideal-img/[name].[hash:hex:7].[width].[ext]'
                                            : 'ideal-img/[name].[width].[ext]' }, options),
                                },
                            ],
                        },
                    ],
                },
            };
        },
    };
}
exports.default = default_1;
